### PR TITLE
Egg/player interactions

### PR DESCRIPTION
**Pull request by camillewalters**
To test, enable EggSpawner by uncommenting relevant sections. Pick up eggs by running over them. Throw by clicking. 
To test dropping (roughly, just to test motion), uncomment line 122 and comment line 123 in PlayerController.cs. Then pick up eggs, and click to drop.

Things I need feedback on:

- How the eggs stack up on top of the player (there's probably a better way of doing this?)
- Do we like the throwing/dropping motion?
- Is the area that the eggs can be picked up around the player too big?

Closes https://github.com/camillewalters/Eggstravaganza/issues/20

### CODE REVIEW

**Review by priyanka-cs — COMMENTED** [review]
Works great so far its so cool to see it coming together! 🎉 
I agree with Amanda, the egg stacking, throwing and dropping is very neat. I do think that the  area that the eggs can be picked up around the player might be too big, but I think making the pickup forward facing will make it more intuitive, so we can go from there.

I did notice one thing though, idk if its just on my system but the player couldn't pick up the eggs it had dropped. Is that something we want?

**Comment by stephaniewang99** [comment]
When an egg is thrown, should we just, for simplicity sake, destroy the egg? Because if you throw it off the arena, it kinda still exists there. Everything else looks really smooth tho! Great job! 👍 

Side note: I think maybe the trajectory of throwing the egg should go from the bottom rather than the top? because if your egg tower is a lot taller than someone else's, I don't think it's possible to hit another player because it gets thrown too far up? I'm not sure

**Comment by at669** [comment]
> When an egg is thrown, should we just, for simplicity sake, destroy the egg? Because if you throw it off the arena, it kinda still exists there. 

i think we can allow eggs to be thrown off the arena and make a quick catch-all collider to Destroy any infinitely falling eggs! in a stretch goal i could imagine this being a silly stat in the end game screen like "player who threw most eggs into the void"

**Comment by priyanka-cs** [comment]
> > When an egg is thrown, should we just, for simplicity sake, destroy the egg? Because if you throw it off the arena, it kinda still exists there.
> 
> i think we can allow eggs to be thrown off the arena and make a quick catch-all collider to Destroy any infinitely falling eggs! in a stretch goal i could imagine this being a silly stat in the end game screen like "player who threw most eggs into the void"

i LOVE THIS

File: Eggstravaganza/Assets/Scripts/EggSpawn.cs [hunk 1]
```diff
@@ -20,7 +20,7 @@ public class EggSpawn : MonoBehaviour
 
     [SerializeField]
     float spawnAreaScale = 1;
-    
+
```
[the remark this answers is not in the document]
Comment (camillewalters) [reply Eggstravaganza/Assets/Scripts/EggSpawn.cs:23]: i use visual studio. i wonder if its a line ending thing? idk

**Review by at669 — COMMENTED** [review]
at player speed ~120 and drag ~10, the movement feels nice :+1: 

should be interesting to network the "getting hit by egg" experience later!

last couple of questions: 

* when i drop an egg, it currently goes through the floor—is this the expected behavior?
* there's some conflicts with EggSpawn and the PlayerPrefab; i don't think there should be any changes to EggSpawn since it's just whitespace changes, but could you pull and re-apply changes to the PlayerPrefab?
* could you push your speed, drag, and components to the Player prefab as well after pulling in latest changes?

File: Eggstravaganza/Assets/Prefabs/Player.prefab [hunk 2]
```diff
@@ -395,7 +428,20 @@ MonoBehaviour:
   speed: 5
   lerpParam: 10
   body: {fileID: 3361834574544056098}
-  eggInventory: 
+  eggInventory: []
+--- !u!65 &1231281954442966026
+BoxCollider:
```
[the remark this answers is not in the document]
Comment (camillewalters) [reply Eggstravaganza/Assets/Prefabs/Player.prefab:547]: Could you clarify what you mean by the Egg boxcollider (if this is still an issue)?

File: Eggstravaganza/Assets/Scripts/EggSpawn.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (at669) [reply Eggstravaganza/Assets/Scripts/EggSpawn.cs]: i think usually unity starts yelling about line endings if they're mismatched, maybe it's just a whitespace/tabs vs spaces setting in the IDE. something to investigate later maybe!

**Review by priyanka-cs — COMMENTED** [review]
with a higher speed the movement is great! idk why I still see the problem where if I drop an egg I can't pick it up again, maybe its a just a prob on my end? If anyone else doesn't see it we're good to go

**Review by priyanka-cs — COMMENTED** [review]
I can pick dropped eggs now yay! some eggs fall through the ground, but its very few we could ignore it
One last thing that I missed before, for some reason my gopher can't move for a few seconds after it loses an egg. Is that expected?

**Review by picrossin — APPROVED** [review]

**Comment by at669** [comment]
> I can pick dropped eggs now yay! some eggs fall through the ground, but its very few we could ignore it One last thing that I missed before, for some reason my gopher can't move for a few seconds after it loses an egg. Is that expected?

yes that's intentional! you're "stunned" for a second if you get hit (and therefore lose an egg)

**Comment by priyanka-cs** [comment]
> > I can pick dropped eggs now yay! some eggs fall through the ground, but its very few we could ignore it One last thing that I missed before, for some reason my gopher can't move for a few seconds after it loses an egg. Is that expected?
> 
> yes that's intentional! you're "stunned" for a second if you get hit (and therefore lose an egg)

oh shit my bad I forgot about this, then we're good to go yay!

**Review by priyanka-cs — APPROVED** [review]
🎉

**Pull request merged**